TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MecanumTeleOp.java

```diff
@@ -1,43 +1,99 @@
+import com.qualcomm.hardware.kauailabs.NavxMicroNavigationSensor;
+import com.qualcomm.robotcore.hardware.IntegratingGyroscope;
+import com.qualcomm.robotcore.util.ElapsedTime;
+import org.firstinspires.ftc.robotcore.external.navigation.Orientation;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
+
+
 
 @TeleOp
 public class MecanumTeleOp extends LinearOpMode {
     @Override
+
     public void runOpMode() {
-        DcMotor frontLeft = hardwareMap.dcMotor.get("frontLeft");
-        DcMotor backLeft = hardwareMap.dcMotor.get("backLeft");
-        DcMotor frontRight = hardwareMap.dcMotor.get("frontRight");
-        DcMotor backRight = hardwareMap.dcMotor.get("backRight");
+        Hardware hardware = new Hardware(hardwareMap);
+        hardware.backLeft.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        hardware.frontLeft.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        hardware.backRight.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        hardware.frontRight.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        IntegratingGyroscope gyro;
+        NavxMicroNavigationSensor navxMicro;
+        ElapsedTime timer = new ElapsedTime();
+        navxMicro = hardwareMap.get(NavxMicroNavigationSensor.class, "gyro");
+        gyro = (IntegratingGyroscope) navxMicro;
 
-        frontLeft.setDirection(DcMotorSimple.Direction.REVERSE);
-        backLeft.setDirection(DcMotorSimple.Direction.REVERSE);
+        telemetry.log().add("Gyro Calibrating. Do Not Move!");
+
+        // Wait until the gyro calibration is complete
+        timer.reset();
+        while (navxMicro.isCalibrating()) {
+            telemetry.addData("calibrating", "%s", Math.round(timer.seconds()) % 2 == 0 ? "|.." : "..|");
+            telemetry.update();
+        }
+        telemetry.log().clear();
+        telemetry.log().add("Gyro Calibrated. Press Start.");
+        telemetry.clear();
+        telemetry.update();
 
         waitForStart();
         if (isStopRequested()) return;
 
         while (opModeIsActive()) {
-            double y = -gamepad1.left_stick_y;
-            double x = gamepad1.left_stick_x * 1.1;
+
+            Orientation angles = gyro.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.RADIANS);
+            double botheading = angles.firstAngle;
+            telemetry.addLine()
+                    .addData("heading", formatAngle(angles.angleUnit, angles.firstAngle))
+                    .addData("roll", formatAngle(angles.angleUnit, angles.secondAngle))
+                    .addData("pitch", "%s deg", formatAngle(angles.angleUnit, angles.thirdAngle))
+                    .addData("botheading",formatAngle(angles.angleUnit,botheading));
+
+            telemetry.update();
+            double y = gamepad1.left_stick_y; // Remember, Y stick value is reversed
```

Comment:
Comment says `Remember, Y stick value is reversed` but the value isn't negated here. does the math just work out?